app/Http/Middleware/AuthAdmin.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Contracts\Auth\Guard;
+
+class AuthAdmin
+{
+    /**
+     * The auth instance.
+     *
+     * @var Illuminate\Contracts\Auth\Guard
+     */
+    protected $auth;
+    
+    /**
+     * Create a new auth instance.
+     *
+     * @param Illuminate\Contracts\Auth\Guard $auth Auth
+     */
+    public function __construct(Guard $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request request
+     * @param \Closure                 $next    next
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if ($this->auth->guest()) {
+            if ($request->ajax()) {
+                return response('Unauthorized.', 401);
```

Comment:
What is `401`?
Please use `use Symfony\Component\HttpFoundation\Response;` and `Response::HTTP_UNAUTHORIZED` to get response status.